/*
 * Xbase supports Exception handling using the same syntax as is used 
 * in Java. There are two differences:
 * 1) Checked exceptions must not be catched in closures.
 * 2) Try-catch is an expression and can therefore be used in a deeply 
 *    nested way 
 */
try {
	(null as Object).toString
} catch (NullPointerException e) {
	throw new RuntimeException(e)
} finally {
	// do nothing	
}
// threw RuntimeException (String)

/*
 * Note that within purexbase after you've thrown an exception, the 
 * interpreter will always result in the first thrown exception.
 */
null
// threw RuntimeException (Void)



